src/backend/distributed/operations/shard_transfer.c

```diff
@@ -1611,6 +1611,21 @@ EnsureShardCanBeCopied(int64 shardId, const char *sourceNodeName, int32 sourceNo
 							"shard " INT64_FORMAT " already exists in the target node",
 							shardId)));
 	}
+
+	/*
+	 * Make sure the relation exists. In some cases the relation is actually dropped but
+	 * the metadata remains, such as dropping table while citus.enable_ddl_propagation
+	 * is set to off.
+	 */
+	ShardInterval *shardInterval = LoadShardInterval(shardId);
+	Oid distributedTableId = shardInterval->relationId;
+	if (!RelationExists(distributedTableId))
+	{
+		ereport(ERROR, (errmsg("relation with OID \"%u\" does not exist",
```

Comment:
quotes are not needed, since oid is always a number. It's also consistent with some of our other error messages
```suggestion
		ereport(ERROR, (errmsg("relation with OID %u does not exist",
```

Reply:
I was going to do that but then I thought a hint would be very useful because it's a really weird case to hit

Reply:
This kind of stuff can happen for a variety of reasons. The ddl propagation flag is mainly one of them that's accessible by non superusers.